src/test/java/net/starlark/java/eval/MethodLibraryTest.java

```diff
@@ -772,6 +774,29 @@ ev.new Scenario()
             "','.join(elements=['foo', 'bar'])");
   }
 
+  @StarlarkBuiltin(name = "named_only", doc = "")
+  static final class NamedOnly implements StarlarkValue {
+    @StarlarkMethod(
+        name = "foo",
+        documented = false,
+        parameters = {
+          @Param(name = "a"),
+          @Param(name = "b", named = true, defaultValue = "None", positional = false),
+        })
+    public Object foo(Object a, Object b) {
+      return a;
+    }
+  }
+
+  @Test
+  public void testNamedOnlyArgument() throws Exception {
```

Comment:
I added this test to document why I think that https://github.com/bazelbuild/bazel/pull/12782#discussion_r773419757 isn't a safe optimization, but I may have misunderstood the original comment.